pkg/logs/launchers/integration/launcher.go

```diff
@@ -86,58 +118,138 @@ func (s *Launcher) run() {
 			for _, source := range sources {
 				// TODO: integrations should only be allowed to have one IntegrationType config.
 				if source.Config.Type == config.IntegrationType {
-					logFilePath, err := s.createFile(cfg.IntegrationID)
-					if err != nil {
-						ddLog.Warn("Failed to create integration log file: ", err)
-						continue
+					// This check avoids duplicating files that have already been created
+					// by scanInitialFiles
+					logFile, exists := s.integrationToFile[cfg.IntegrationID]
+
+					if !exists {
+						logFile, err = s.createFile(cfg.IntegrationID)
+						if err != nil {
+							ddLog.Warn("Failed to create integration log file:", err)
+							continue
+						}
+
+						// file to write the incoming logs to
+						s.integrationToFile[cfg.IntegrationID] = logFile
 					}
-					filetypeSource := s.makeFileSource(source, logFilePath)
-					s.sources.AddSource(filetypeSource)
 
-					// file to write the incoming logs to
-					s.integrationToFile[cfg.IntegrationID] = logFilePath
+					filetypeSource := s.makeFileSource(source, logFile.filename)
+					s.sources.AddSource(filetypeSource)
 				}
 			}
 
 		case log := <-s.integrationsLogsChan:
-			logFilePath := s.integrationToFile[log.IntegrationID]
+			s.receiveLogs(log)
+		case <-s.stop:
+			return
+		}
+	}
+}
 
-			err := s.ensureFileSize(logFilePath)
-			if err != nil {
-				ddLog.Warn("Failed to get file size: ", err)
-				continue
-			}
+// receiveLogs handles writing incoming logs to their respective file as well as
+// enforcing size limitations
+func (s *Launcher) receiveLogs(log integrations.IntegrationLog) {
+	fileToUpdate := s.integrationToFile[log.IntegrationID]
```

Comment:
nit: instead of checking `fileToUpdate == nil`, I think it would be better to check if the element exists by 
```
fileToUpdate, exists := s.integrationToFile[log.IntegrationID]
if !exists {
    return 
}
```

Reply:
Changed